src/graphql/index.ts

```diff
@@ -0,0 +1,25 @@
+import { gql, Config } from "apollo-server";
+import { DocumentNode } from "graphql";
+import { registerNewCart } from "../cart/cart";
+import { productsResolvers } from "./product/product.resolver";
+import { productsSchema } from "./product/product.schema";
+import { ProductTypeDef } from "./product/product.types.schema";
+
+const baseSchema: DocumentNode = gql`
+  type Query
+  type Mutation
+  type Subscription
+`
+
+export const apolloServerConfig: Config = {
+    typeDefs: [baseSchema, productsSchema, ProductTypeDef],
+    resolvers: productsResolvers,
+    context: (context) => {
```

Comment:
this is called every time and you go around it by checking if it exists inside registerNewCart.

for simplicity you may assume that every user will subscribe to productsUpdate, and you can register the cart when the subscription is made

Reply:
R u sure is it the best way?